tests/workflow.go

```diff
@@ -980,3 +983,74 @@ func (s *FunctionalSuite) TestWorkflowRetryFailures() {
   4 WorkflowTaskCompleted
   5 WorkflowExecutionFailed`, events)
 }
+
+func (s *FunctionalSuite) TestExecuteMultiOperation() {
+	s.Run("Start Workflow and send Update", func() {
+		tv := testvars.New(s.T().Name())
+
+		request := &workflowservice.ExecuteMultiOperationRequest{
+			Namespace: s.namespace,
+			Operations: []*workflowservice.ExecuteMultiOperationRequest_Operation{
+				{
+					Operation: &workflowservice.ExecuteMultiOperationRequest_Operation_StartWorkflow{
+						StartWorkflow: &workflowservice.StartWorkflowExecutionRequest{
+							RequestId:    uuid.New(),
+							WorkflowId:   tv.WorkflowID(),
+							WorkflowType: tv.WorkflowType(),
+							TaskQueue:    tv.TaskQueue(),
+							Input:        nil,
+							Identity:     tv.WorkerIdentity(),
+						},
+					},
+				},
+				{
+					Operation: &workflowservice.ExecuteMultiOperationRequest_Operation_UpdateWorkflow{
+						UpdateWorkflow: &workflowservice.UpdateWorkflowExecutionRequest{
+							Request: &updatepb.Request{
+								Meta:  &updatepb.Meta{UpdateId: "UPDATE_ID"},
+								Input: &updatepb.Input{Name: "UPDATE"},
+							},
+							WorkflowExecution: &commonpb.WorkflowExecution{WorkflowId: tv.WorkflowID()},
+							WaitPolicy:        &updatepb.WaitPolicy{LifecycleStage: enumspb.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_COMPLETED},
+						},
+					},
+				},
+			},
+		}
+
+		poller := &TaskPoller{
+			Engine:    s.engine,
+			Namespace: s.namespace,
+			TaskQueue: tv.TaskQueue(),
+			Identity:  tv.WorkerIdentity(),
+			WorkflowTaskHandler: func(execution *commonpb.WorkflowExecution, wt *commonpb.WorkflowType, previousStartedEventID, startedEventID int64, history *historypb.History) ([]*commandpb.Command, error) {
+				return nil, nil
+			},
+			MessageHandler: func(task *workflowservice.PollWorkflowTaskQueueResponse) ([]*protocolpb.Message, error) {
+				updRequestMsg := task.Messages[0]
+				return s.UpdateAcceptCompleteMessages(tv, updRequestMsg, "1"), nil
+			},
+			Logger: s.Logger,
+			T:      s.T(),
+		}
+
+		resultCh := make(chan *workflowservice.ExecuteMultiOperationResponse)
+		go func() {
+			resp, err := s.engine.ExecuteMultiOperation(NewContext(), request)
+			s.NoError(err)
+			resultCh <- resp
+		}()
+
+		_, err := poller.PollAndProcessWorkflowTask(WithDumpHistory)
+		s.NoError(err)
+
+		resp := <-resultCh
```

Comment:
You should use a context with a timeout here so that a stall in our ExecuteMultiOperation channel won't cause our tests to hang. Something like
```
ctx, cancel := context.WithTimeout(context.Background(), 30 * time.Second) // TODO set reasonable timeout

select {
    case <- ctx.Done():
        s.Fail("Timed out waiting for result of multioperation")
        return
    case: resp: <-resultCh:
        ...
}
```

We do this in a few other places for this reason, though I cannot find them at the moment